### ISSUE TITLE
Column autoresizing explodes on something triggered by "sudo apt autoremove", Ubuntu 16.04

### ISSUE THREAD

**Issue by Gotterbild** [Alexey-Yakovenko/deadbeef#1672]
All texts disappeared after "sudo apt autoremove"
Here's screenshot of how player now looks like: https://s.mail.ru/LZKk/kwRainCX6

Tried to change fonts in Settings / Layout / Playlist (Настройки / Оформление / Плейлист) - nothing happens. Screenshot: https://s.mail.ru/Kdvt/E6F3wSWC6

My system is Xubuntu 16.04
Kernel version 4.4.0-47-generic
Screenshot: https://s.mail.ru/HxKo/79V2uCiWs

How that happened. I installed new music players to replace Audacious. They were quodlibet, clementine, deadbeef and qmmp. Cheked out all of them, played with settings a bit, liked Deadbeef most. Then "sudo apt purge audacious quodlibet clementine qmmp", then "sudo apt autoremove". In the process of autoremove all texts in playlist window disapeared.

**Comment by Gotterbild** [comment]
Simple reinstall didn't help. (sudo apt purge deadbeef && sudo apt install deadbeef)

Only helped when I killed .config\deadbeef folder after purging and then installed again.

**Comment by Alexey-Yakovenko** [comment]
I'm not 100% sure, but it seems to be caused by an old bug with column autoresizing feature.
Basically the first column became reeeally large, going far out of the player window width.

Try scrolling to the right, and making the column smaller.

I thought that this bug was fixed long time ago...

It was not reported for quite some time.

[One of these bugs](https://github.com/Alexey-Yakovenko/deadbeef/issues?q=is%3Aissue+autoresize+is%3Aclosed)

**Comment by Alexey-Yakovenko** [comment]
It seems that all of the above bugs were fixed on the 0.7 branch before the 0.7.2 release.

So that seems to be a new way to trigger the issue.. Maybe a GTK theme reload caused this, or something like that.

**Comment by cboxdoerfer** [comment]
@Gotterbild, do you have (or was) `Preferences -> GUI/Misc -> Playlist -> Auto-resize columns to fit the window` enabled? If so, does the bug still happen when you disable it?

**Comment by Gotterbild** [comment]
I can't say if auto-resize option was cheked when issue appeared. But I confirm that I tried to enable this option, but nothing was happening. As I said higher, I've reinstalled deadbeef which solved the issue. Right now I activated this option and nothing happened. Columns didn't changed their size, issue didn't appear again.

Look at first screenshot - there is no horizontal scrollbar. So maybe it is not connected with column width issue.

**Comment by Alexey-Yakovenko** [comment]
@Gotterbild it's not enough to reinstall the player or change the settings to trigger the issue again. You'd need to perform the same steps, that caused it the first time -- i.e. run apt autoremove, and probably whatever other things happened before.

**Comment by Alexey-Yakovenko** [comment]
> Look at first screenshot - there is no horizontal scrollbar. So maybe it is not connected with column width issue.

I noticed that too, but in Ubuntu you never know for sure.

Either way, the only way to find out what really happened was to have a look at your configuration file. But you deleted it before I could even ask for it :)

**Comment by Gotterbild** [comment]
> I noticed that too, but in Ubuntu you never know for sure.
I have Xubuntu and in my theme scrollbars are always visible. When I resize one of the columns to be too big, horizontal scrollbar appears right above status bar: https://s.mail.ru/DHAH/dVVr1JAti

> you deleted it before I could even ask for it :)
Wanted to listen to music RIGHT NOW, sorry :) 

> You'd need to perform the same steps, that caused it the first time -- i.e. run apt autoremove, and probably whatever other things happened before.
Hmm. Tried to experiment a bit to recreate what I did before.
1) Cheked auto-resize columns option: https://s.mail.ru/4cg8/Up2VKpD7f
2) Played with Settings / GUI / Playlist https://s.mail.ru/JkUr/yqg7H1L7G (I did that before issue appeared)
3) sudo apt install audacious qmmp clementine quodlibet (after unpacking ~63 MB)
4) sudo apt purge audacious qmmp clementine quodlibet (freed ~32 MB)
5) sudo apt autoremove (freed ~31MB)
While all this actions Deadbeef was opened full screen. Then I opened Terminal window (Deadbeef still visible in background) and ran the commands.
No issue appeared.

Then I played with changing OSes GTK themes while Deadbeef was opened full screen in the background (still visible, just out of focus). No issues.

Today I have no time to continue experiments. But if you ask, tomorrow I can try to fully recreate what I can - kill deadbeef with configs, install only audacious, then install all that players including deadbeef, play with all players, then play with deadbeef's settings and then purge and autoremove.

**Comment by Alexey-Yakovenko** [comment]
It would be nice if you could re-create the issue, and send the configuration file after it happens.

The screenshot actually looks like something I have not seen before.

It looks like there are no columns at all in the playlist, but some numbers are still drawn in the playlist. These numbers look like indexes of playlist items. They're not even displayed by default, so I suppose that's something you customized before the issue happened.

**Comment by revyh** [comment]
Hello! It looks like I have same issue. Playlist also doesn't have horizontal scrollbars. Tried to toggle `Preferences -> GUI/Misc -> Playlist -> Auto-resize columns to fit the window` and both GTK2/GTK3 themes, but with no luck. My system is xubuntu 16.10. @Alexey-Yakovenko what configs do you need to further investigate the issue?

![screenshot_2017-01-18_10-20-36](https://cloud.githubusercontent.com/assets/668565/22049709/2cb94b0c-dd70-11e6-9e8a-f9d6ee399bfe.png)

**Comment by Alexey-Yakovenko** [comment]
@revyh ~/.config/deadbeef/config

**Comment by revyh** [comment]
add `.txt` extension just because github can't upload files without exntension

[config.txt](https://github.com/Alexey-Yakovenko/deadbeef/files/713645/config.txt)

**Comment by Alexey-Yakovenko** [comment]
@revyh thanks a lot for the config. It really shows the problem. It's not caused by what I thought. Seems like there's some bug, which corrupts the configuration file, by adding way to many columns, which can't be saved correctly. I think this issue can be reproduced by creating many columns (about 20), then restarting the player.

**Comment by revyh** [comment]
@Alexey-Yakovenko When the problem appeared, I only saw the first two columns: "Cover" and "♫". I tried to "fix" this by adding new columns and trying to recreate the state of the columns, which I had before. But when I d'n'd new columns after "♫", they have "disappeared". So I don't think that something is added too many columns in the config (in `gtkui.columns.playlist` option if I understood you well). I think that it was me.

After your last comment, I tried to delete entire `gtkui.columns.playlist` option and deadbeef recreated it correctly (based on `playlist.column` options I think) after restart and my playlist looks well now. Then I set big negative "size" (-10000) in third column in `gtkui.columns.playlist` option and it reproduced the issue. It seems that negative size moves column out of viewport and d'n'd columns "disappeared" because they are placed before this "bad" column (looks like their become negative sized too).

But negative size should be only consequences of the real problem. I can't remember what I did before the issue happened, but "something gone wrong", maybe "apport" reported about some crash. I also noticed that first column with negative size (from the config that I post) also has `"id": "0"` which doesn't correspond to any id in `playlist.column` options (third number from the end, if my guess is correct). I tried to "corrupt" ids in `gtkui.columns.playlist` but it didn't affect anything.
